llm/memory.go

```diff
@@ -437,3 +450,39 @@ func projectorMemoryRequirements(filename string) (weights, graphSize uint64) {
 
 	return weights, graphSize
 }
+
+// estimateKvCacheSize determines the memory required for K or V cache based on the quantization type
+func estimateKvCacheSize(cacheType string, numCtx, blockCount, embeddingHeadCount, headCountKV uint64, isEmbeddingModel bool) uint64 {
```

Comment:
I think simplifying this down to one value for `k` and `v` might reduce the need for this function and we can do it inline

Reply:
Although I'm okay to keep it for testing purposes if it's helpful there

Reply:
Won't that mean that if your keys are Q4_0 but your values are Q8_0 (or f16 even) - you'll be estimating based on the wrong information?

Reply:
Updated to force the k and v quantisation types to be the same.

Reply:
FYI updated the version where you can also set this in the Modelfile with the combined k/v as well if you end up wanting to add it in - https://github.com/sammcj/ollama/pull/63/files